mongodbatlas/resource_mongodbatlas_search_index.go

```diff
@@ -48,9 +47,10 @@ func returnSearchIndexSchema() map[string]*schema.Schema {
 			Required: true,
 		},
 		"analyzers": {
-			Type:     schema.TypeList,
+			Type:     schema.TypeString,
```

Comment:
can this somehow be validated to see if there's a way to validate if it's a working JSON, can you check what AWS does in iam policies for example

Reply:
I think this is achieved when the resource internally try to unmarshall the string into a map of interfaces to verify if it is different from state (this return an error when it cannot be unmarshalled) 

```
func validateSearchAnalyzersDiff(k, old, newStr string, d *schema.ResourceData) bool {
	var j, j2 interface{}

	if old == "" {
		old = "{}"
	}

	if newStr == "" {
		newStr = "{}"
	}

	if err := json.Unmarshal([]byte(old), &j); err != nil {
		log.Printf("[ERROR] cannot unmarshal old search index analyzer json %v", err)
	}
	if err := json.Unmarshal([]byte(newStr), &j2); err != nil {
		log.Printf("[ERROR] cannot unmarshal new search index analyzer json %v", err)
	}
	if diff := deep.Equal(&j, &j2); diff != nil {
		log.Printf("[DEBUG] deep equal not passed: %v", diff)
		return false
	}

	return true
}
```